caffe2/contrib/aten/aten_op_template.h

```diff
@@ -94,11 +94,13 @@ class ATenOp : public Operator<Context> {
     std::vector<int64_t> dims(at_sizes.begin(),at_sizes.end());
     dst->Resize(dims);
     dst->ShareExternalPointer(
-        src.data_ptr(), typeMetaFor(src), 0, [src](void* ptr) mutable {
-          // return a closure that holds a handle to t until it is called
-          // to keep the aten memory alive
-          return src.reset();
-        });
+        at::DataPtr(
+            (void*)src.data_ptr(),
+            static_cast<void*>(&src),
```

Comment:
This looks wrong. If you do this, you're going to attempt to `delete` a pointer to a stack allocated variable. That's very bad juju. You *need* to do a dynamic allocation here, if you want to make this work.